### DISCUSSION TITLE
Custom CSS on Conditional UI in Repeater

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#8003] [Help]
I have the following scenario
 
Data Provider 
- Repeater
- - Container 

Container -> OnClick -> Set State Variable "salectedID = Repeater.id"
Container -> Conditional UI -> Set Custom CSS to "border: 1px solid blue !important" if Repeater.id === selectedID

Now the outcome is that the "clicked" container gets the custom CSS but the custom CSS is not removed from previously clicked containers.

See below 

https://user-images.githubusercontent.com/80512170/192583114-08ff2326-2ed4-48e2-aa8a-965716aa252e.mov

**Comment by melohagan** [comment]
Hey @poirazis 

Is this not expected? If you wanted to make sure only one row can be selected at a time, then you can add an additional condition:

![Screenshot 2022-09-28 at 08 41 11](https://user-images.githubusercontent.com/101575380/192718739-04706fe1-36c9-413e-a3a6-85a06532b5de.png)

As long as you use pagination for many rows then performance won't be an issue. I tried it out with a limit of 50 and it ran smoothly.

**Reply by poirazis** [reply]
Hey @melohagan 

up until now, i was using the conditional UI as though as if it had a default "ELSE do the Default behavior"
for example , when i used a conditional UI to make a Button Large, i didn't use the second rule ( IF NOT )
it was UPDATE SETTING SET SIZE=LARGE IF X=Y,  without a rule second rule.
( I have used this widely in my "tab control" implementation with buttons , the selected tab becomes large )

I think @aptkingston mentioned this somewhere

**Reply by melohagan** [reply]
Yeah that is true. This case might be a bit different in how it works under the hood compared to other conditional UI, but I'm no sure myself.
